qp_klp/SequencingTech.py

```diff
@@ -0,0 +1,256 @@
+        # properly. Append these to the failed-samples report and also
+        # return the list directly to the caller.
+        failed_samples = job.audit(self.pipeline.get_sample_ids())
+        if hasattr(self, 'fsr'):
+            # NB 16S does not require a failed samples report and
+            # it is not performed by SPP.
+            self.fsr.write(failed_samples, job.__class__.__name__)
+
+        return failed_samples
+
+
+class TellSeq(SequencingTech):
+    seqtech_type = SEQTECH_NAME_TELLSEQ
+
+    def convert_raw_to_fastq(self):
+        config = self.pipeline.get_software_configuration('tell-seq')
+
+        job = TellReadJob(self.pipeline.run_dir,
+                          self.pipeline.output_path,
+                          self.pipeline.input_file_path,
+                          config['queue'],
+                          config['nodes'],
+                          config['wallclock_time_in_minutes'],
+                          config['tellread_mem_limit'],
+                          config['modules_to_load'],
+                          self.master_qiita_job_id,
+                          config['reference_base'],
+                          config['reference_map'],
+                          config['sing_script_path'],
+                          config['tellread_cores'])
+
+        self.raw_fastq_files_path = join(self.pipeline.output_path, 'Full')
+
+        job.run(callback=self.status_update_callback)
+
+        self.pipeline.get_sample_ids()
+        failed_samples = []
+        if hasattr(self, 'fsr'):
+            # NB 16S does not require a failed samples report and
+            # it is not performed by SPP.
+            self.fsr.write(failed_samples, job.__class__.__name__)
+
+        return failed_samples
+
+    def generate_norm_counts(self):
+        config = self.pipeline.get_software_configuration('tell-seq')
+
+        job = TRNormCountsJob(self.pipeline.run_dir,
+                              self.pipeline.output_path,
+                              self.pipeline.input_file_path,
+                              config['queue'],
+                              config['nodes'],
+                              config['wallclock_time_in_minutes'],
+                              config['normcount_mem_limit'],
+                              config['modules_to_load'],
+                              self.master_qiita_job_id,
+                              'foo',
+                              config['integrate_script_path'],
+                              self.qiita_job_id)
+
+        job.run(callback=self.status_update_callback)
+
+        # audit the results to determine which samples failed to convert
+        # properly. Append these to the failed-samples report and also
+        # return the list directly to the caller.
+        failed_samples = job.audit(self.pipeline.get_sample_ids())
+        if hasattr(self, 'fsr'):
+            # NB 16S does not require a failed samples report and
```

Comment:
We have not considered use of TellSeq for 16S

Reply:
Gotcha, I'll remove the note from this class.